dpnp/dpnp_iface_arraycreation.py

```diff
@@ -901,16 +913,79 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis
 
     """
 
-    if not use_origin_backend():
-        if axis != 0:
-            checker_throw_value_error("linspace", "axis", axis, 0)
+    if retstep is not False:
+        pass
+    elif axis != 0:
+        pass
+    elif dpnp.isscalar(start) and dpnp.isscalar(stop):
+        usm_type = usm_type if usm_type is not None else "device"
+        dt = None if numpy.issubdtype(dtype, dpnp.integer) else dtype
+        res = dpnp_container.linspace(start,
+                                       stop,
+                                       num,
+                                       dtype=dt,
+                                       device=device,
+                                       endpoint=endpoint,
+                                       sycl_queue=sycl_queue,
+                                       usm_type=usm_type)
+        if numpy.issubdtype(dtype, dpnp.integer):
+            dpnp.floor(res, out=res)
+            res = res.astype(dtype)
+        return res
+    else:
+        start_isarray = dpnp.isarray(start)
+        stop_isarray = dpnp.isarray(stop)
+
+        if start_isarray and stop_isarray:
+            dt = numpy.result_type(start.dtype, stop.dtype)
+            if sycl_queue is None and device is None:
+                sycl_queue = dpnp.get_execution_queue([start.sycl_queue, stop.sycl_queue])
+            if usm_type is None:
+                usm_type = dpnp.get_coerced_usm_type([start.usm_type, stop.usm_type])
+
+        elif start_isarray:
+            dt = start.dtype
+            if sycl_queue is None and device is None:
+                sycl_queue = start.sycl_queue
+            usm_type = start.usm_type if usm_type is None else usm_type
+        elif stop_isarray:
+            dt = stop.dtype
+            if sycl_queue is None and device is None:
+                sycl_queue = stop.sycl_queue
+            usm_type = stop.usm_type if usm_type is None else usm_type
+        else:
+            dt = numpy.result_type(None)
 
-        res = dpnp_linspace(start, stop, num, endpoint, retstep, dtype, axis)
+        if numpy.issubdtype(dt, dpnp.integer):
+            dt = numpy.result_type(float(num), dt)
+        usm_type = usm_type if usm_type is not None else "device"
+        sycl_queue_normalized = dpnp.get_normalized_queue_device(sycl_queue=sycl_queue, device=device)
 
-        if retstep:
-            return res
-        else:
-            return res[0]
+        _start = array([start] if dpnp.isscalar(start) else start, dtype=dt,
+                       usm_type=usm_type, sycl_queue=sycl_queue_normalized)
+        _stop = array([stop] if dpnp.isscalar(stop) else stop, dtype=dt,
+                      usm_type=usm_type, sycl_queue=sycl_queue_normalized)
+
+        if dtype is None:
+            dtype = dt
+
+        step = ((_stop - _start) * (1. / (num - 1) if endpoint else num)).astype(dt)
+
+        res = dpnp_container.arange(0,
```

Comment:
Since we are not going to use dpctl here (when both start and stop aren't scalar), we must to validate input arguments: their type and range.